### DISCUSSION TITLE
mini.completion: Is it feasible to increase locality strength?

### DISCUSSION THREAD

**Discussion by wroyca** [nvim-mini/mini.nvim#380] [Q&A] [answered]
Greetings! 

An area I want to customize is mini.completion locality since it is presently insufficient in my workflow. In the screenshot provided below, I expected "operation_rule_map" to be the primary suggestion considering its locality, but disappointingly, it does not appear — even after inputting a few extra letters:

![Screenshot from 2023-06-22 22-54-38](https://github.com/echasnovski/mini.nvim/assets/91024200/afa6eafd-4845-4acd-b13c-6b8003c59e21)

For comparison, below is a nvim-cmp screenshot, where only the buffer and LSP—which is why I want to switch to a lightweight alternative such as mini.completion—are used as sources:

![image](https://github.com/echasnovski/mini.nvim/assets/91024200/77276947-13ea-4789-a679-d045804d7528)

As depicted, the results are sorted based on locality, with "operation_rule_map" promptly appearing as the initial suggestion - something I'd like to achieve with mini.completion. 

Is there a particular configuration that could enhance this aspect? :coffee:

**Comment by echasnovski** [comment, marked as answer]
Hi!

What you are seeing in 'mini.completion' screenshot is the fallback builtin completion. By default it is the same as pressing `<C-n>` without completion engine. You can try other keys (see `:h ins-completion`) in `fallback_action` config setting. I would assume that `'<C-p>'` should work in this case, but it doesn't seem like it. So not sure how to make it happen here (apart from relying on LSP server for more proper suggestions).

Other than built-in completion or fully user-defined function fully responsible for showing completion menu, nothing else is allowed in

**Reply by wroyca** [reply]
> So not sure how to make it happen here (apart from relying on LSP server for more proper suggestions).

Note that LSP is not enabled in my screenshots - I have my own method of enabling it whenever I deem it necessary - so the completion is not relying on it
